File: backend/app/schemas/player.py
from datetime import date, datetime

from pydantic import BaseModel


class PlayerCreate(BaseModel):
    full_name: str
    gender: str | None = None
    birth_date: date | None = None
    department: str | None = None
    phone: str | None = None


class PlayerUpdate(BaseModel):
    full_name: str | None = None
    gender: str | None = None
    birth_date: date | None = None
    department: str | None = None
    phone: str | None = None


class PlayerResponse(BaseModel):
    id: int
    full_name: str
    gender: str | None = None
    birth_date: date | None = None
    department: str | None = None
    status: str
    created_at: datetime | None = None

File: frontend/shared/types/player.ts
import type { Gender, PlayerStatus } from './enums'

export interface Player {
  id: number
  full_name: string
  gender?: Gender
  birth_date?: string
  department?: string
  status: PlayerStatus
  created_at: string
  updated_at: string
}

export interface PlayerCreate {
  full_name: string
  gender?: Gender
  birth_date?: string
  department?: string
}

export interface PlayerUpdate {
  full_name?: string
  gender?: Gender
  birth_date?: string
  department?: string
  status?: PlayerStatus
}
